community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/state/NeoStoreFileListing.java

```diff
@@ -66,77 +75,56 @@ public NeoStoreFileListing( File storeDir, LabelScanStore labelScanStore,
                 new MultiResource( asList( labelScanStoreSnapshot, schemaIndexSnapshots, legacyIndexSnapshots ) ) );
     }
 
-    private Resource gatherLegacyIndexFiles( Collection<File> files ) throws IOException
+    private void gatherNonRecordStores( Collection<StoreFileMetadata> files, boolean includeLogs )
+    {
+        for ( File file : storeDir.listFiles() )
```

Comment:
The `File.listFiles()` is actually allowed to return `null` if the given file is not a directory, or if an IO error occurs. We need to check for that before iterating. I think a `null` return value should be treated the same as an empty array in this case.